doc/sphinx/proofs/writing-proofs/proof-mode.rst

```diff
@@ -4,6 +4,87 @@
  Proof handling
 -------------------
 
+.. todo PR: my example is pretty bad--can you suggest a better one?
+   Also need to position/integrate this better
+
+Proof State
+-----------
+
+Proof state consists of one or more subgoals that need to be proven.
+Each subgoal has a :gdef:`conclusion` (the statement that is to be proven)
```

Comment:
The wording in the output says we have subgoals rather than contexts, which sure look like they include hypotheses:

```
1 subgoal
  
  ============================
  forall n m : nat, n > m -> P 1 /\ P 2
```

"context" has 464 uses in the rsts, but I think it's for multiple meanings and it's such a generic word.  I think it's a poor choice to use all over the documentation, though in the code, we refer to the  "proof context" with a good bit of precision.

There are 122 occurrences of "conclusion" in the rsts, some of which (I didn't count them) are precisely as I use it, e.g. the first sentence of the tactics chapter: "A deduction rule is a link between some (unique) formula, that we call the *conclusion* and (several) formulas that we call the *premises*."  I think this reflects the terminology people use regularly--I think it's better not to be prescriptive with a single definition in this case.

I think it's not always possible or desirable to have a single meaning for a term (recent example: "equality").  We could just note that "conclusion" can refer to `B` in `A -> B`, as I noted the multiple meanings for `goal` in this subsection.  Again, I think it's following how people use the term.

WDYT?

Reply:
Also, I think there's not really a useful distinction between "goals" and subgoals", so I'd suggest we only use "goals".  AFAIK there is no superior/inferior relationship between goals.  What do you think?

Reply:
OK, you are right that the current practice uses the term conclusion in this sense a lot (though to me the case of the intro of the tactics chapter is not a good example of this because it can be interpreted in both ways and my interpretation differs from yours).

About goals and subgoals, I agree with your suggestion. To me "subgoals" are "goals". The only place I would use "subgoals" is in the following kind of phrase "tactic X transforms the current goal into several subgoals".

Reply:
Who would be a good reviewer for https://github.com/coq/coq/pull/13656? Should I split it up to make it easier to review?

Reply:
I think it is relatively easy to review. It's mostly a matter of making a choice so this PR could probably be raised in an upcoming Coq Call. If you mark the PR as ready to review, you can see if code owners react to it or not...